### ISSUE TITLE
Print to browser - ZIM doesn't use the default browser

### ISSUE THREAD

**Issue by waldauf** [zim-desktop-wiki/zim-desktop-wiki#1340]
Hello,

a few days ago I switched from Firefox to Vivaldi. But when I click on Print to browser the page is opened in Firefox instead of Vivaldi. I tried to set plugin Print to the browser but it doesn't have any options. On the web page, I found the possibility to set Default Applications. But in the _setting / applications_ I found just possible to set editor. 

Next what I tried to search the browser string of Firefox in the config files. But without success. 

Please, is there a way how to say Zim to use the default browser instead of Firefox (another one than default)?

**Comment by pfalcon** [comment]
What makes you think that you have Vivaldi as the default browser? And most importantly - what kind of system do you have?

**Comment by waldauf** [comment]
Damn you're right - I forgot to write what I'm running:
OS: Arch Linux
WM: KDE, where I set Vivaldi as the default browser:

Verified in the terminal
```
❯ xdg-settings get default-web-browser
vivaldi-stable.desktop
```

**Comment by pfalcon** [comment]
For avoidance of doubt, which browser is opened when you run: `xdg-open https://google.com/` ?

**Comment by waldauf** [comment]
`xdg-open https://google.com/` --> Vivaldi

**Comment by jaap-karssenberg** [comment]
You can change the browser Zim sees as the default by trying a weblink in a
page and using the "open with" context menu.

Agree it is a bit obscure.

Regards,

Jaap

Op zo 20 dec. 2020 22:45 schreef Jiri Valnoha <notifications@github.com>:

**Comment by waldauf** [comment]
@jaap-karssenberg Your workaround works like a charm. The first time I had to click "Open with" and chose Vivaldi. And now every link is opened in Vivaldi. Many thx for your answer. :] 

For me is the problem solved. 

Who should close this issue?

**Comment by waldauf** [comment]
I'm closing this issue. Feel free to open it if you need it.

**Comment by waldauf** [comment]
Hello, 

I reopened this issue because I found out about new behavior in the ZIM. If I open a URL from the wiki page it is opened in the default browser after applying the advice from @jaap-karssenberg. Now I tried to export a wiki page to the browser via File / Open in the browser.  For me, it means it is not opened in the default browser (Vivaldi) but in the previous default browser (FF). 

Please, how to fix it?

**Comment by jaap-karssenberg** [comment]
Hi Jiri,

The difference is probably due to opening a local file vs an URL. Have your
tested "xdg-open ./test.html" to check the default browser used for local
HTML files?

As technical background: zim does use the python "webbrowser" module as the
fallback for opening URL's and also for the "print to browser" plugin. This
module in turn tries "xdg-open", followed by Gnome & KDE openers, followed
by any known webbrowsers that might be installed.

So if xdg-open is installed and does the right thing, the plugin should use
the same default.

Regards,

Jaap

**Comment by jaap-karssenberg** [comment]
P.S. working on a fix for #675 which I assume should also take care of this
issue

**Comment by waldauf** [comment]
@jaap-karssenberg You're right. It was caused by different file association for HTML files. I fixed it in KDE. Now it works like charm.

Thanks for your answer.
